traits/has_traits.py

```diff
@@ -812,8 +812,20 @@ def observe_decorator(handler):
             observe_inputs = []
             handler._observe_inputs = observe_inputs
 
+        # Handle the overloaded signature.
+        # Support list to be consistent with on_trait_change.
+        if isinstance(expression, list):
+            expressions = expression
+        else:
+            expressions = [expression]
+
+        expressions = [
+            observe_api.parse(expr) if isinstance(expr, str) else expr
+            for expr in expressions
+        ]
```

Comment:
If this is the approach we want to go for, I'd suggest refactoring this block of logic into a separate function so that both `HasTraits.observe` and the decorator can apply the same logic. This would ensure consistency: If the normalization logic needs to change, they have to change together.